Makefile

```diff
@@ -0,0 +1,55 @@
+SHELL = /bin/bash
+BIN_DIR = ${GOPATH}/bin
+BIN_NAME = ag
+PLUGIN_FOLDER = ${HOME}/.kube/plugins/ag
+.DEFAULT_GOAL = all
+
+.PHONY: clean
+clean:
+	@go clean -testcache || true
+	@rm -rf bin
+
+.PHONYY: gen/plugin
+gen/plugin:
+	hack/gen/plugin.sh
+
+.PHONY: code/check
+code/check:
+	@diff -u <(echo -n) <(gofmt -d `find . -type f -name '*.go' -not -path "./vendor/*"`)
+
+.PHONY: code/fix
+code/fix:
+	@gofmt -w `find . -type f -name '*.go' -not -path "./vendor/*"`
+
+.PHONY: test/unit
+test/unit:
+	@go test -v -race -cover ./pkg/...
+
+.PHONY: test/e2e
+test/e2e:
+	@kubectl apply -f deploy/crds
+	@go test -v -race -cover ./test/e2e/...
+
+.PHONY: test
+test: test/unit test/e2e
+
+.PHONY: build/cli
+build/cli:
```

Comment:
Should this also consider the user's enviroment and build for that? For example, if I run `build/cli` on a Mac, will it work?

Reply:
bin here is the local bin folder, other paths are using GOPATH and HOME env vars which should be ok (I don't have a mac so I need a mac user to test it)